csm/core/controllers/information.py

```diff
@@ -77,3 +77,165 @@ async def post(self):
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}")
+class ViewTopology(CsmView):
+    """
+    GET: Get information about specific resource from deployment topology.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service =self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about all
+           views of specific resource from deployment topology."""
+        Log.info(f"Processing request: {self.request.method} {self.request.path}")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        view = self.request.match_info[const.ARG_VIEW]
+        # Check for valid Resource
+        if resource not in const.TOPOLOGY_RESOURCES:
+            raise CsmNotFoundError(f"Not a valid resource: {resource}")
+        # Check for valid View
+        if view not in const.TOPOLOGY_VIEWS:
+            raise CsmNotFoundError(f"Not a valid view:{view}")
+        # Call Information Service
+        Log.info(f"Fetching deployment topology for resource:{resource}, resource_id:{resource_id} and view:{view}.")
+        try:
+            response = await self._service.get_views(resource, resource_id, view)
+        except CsmInternalError as e:
+            raise e
+        except Exception as e:
+            Log.error(f'{const.UNKNOWN_ERROR}: e')
+            raise CsmInternalError(f"Internal error in fetching topology")
+        Log.info(f"Processed request: {self.request.method} {self.request.path}")
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}/{view_id}")
+class SubviewTopology(CsmView):
+    """GET REST implementation to query information about specific
+       views of specific resource from deployment topology."""
+    Log.info("Handling GET request to fetch details from .")
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query only specific subset of deployment topology."""
+        Log.info(f"Processing request: {self.request.method} {self.request.path}")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        view = self.request.match_info[const.ARG_VIEW]
+        view_id = self.request.match_info[const.ARG_VIEW_ID]
+        # Check for valid Resource
+        if resource not in const.TOPOLOGY_RESOURCES:
+            raise CsmNotFoundError(f"Not a valid resource: {resource}")
+        # Check for valid View
+        if view not in const.TOPOLOGY_VIEWS:
+            raise CsmNotFoundError(f"Not a valid view: {view}")
```

Comment:
Not a valid view: {view} -> Not a valid resource: {resource}

Reply:
done